src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -855,3 +855,46 @@ Tests:
     - "int" : { "$numberLong" :"1"}
     - "int" : { "$numberLong" :"2"}
     - "int" : { "$numberLong" :"3"}
+
+  - Name: Cycle through three pre-computed elements
```

Comment:
@rtimmons what are your thoughts on the naming for the new generators we've added here?

Reply:
Tough one. I think I grok `^Cycle` but `^Once` was a surprise for me. My intuition is to hold off on `^Once` and force the more verbose `{^Cycle: {number:1}}` syntax until we see how often it's used. Just my $0.02.

Reply:
Another thought--the names of the generators may be fine but perhaps rename the keyword args:

```yaml
a: {^Cycle: {ofLength: 3, withValuesGeneratedBy: {^RandomString}}}
```

then just change some internal args to match the public-facing syntax...

I realize it's unordered but phrasing the `ofLength` first makes it a lot more intuitive.

And then `^Once` could be:

```
a: {^FixedGeneratedValue: {fromGenerator: {^FastRandomString}}
```

idk. Bikeshedding is fun. I'll stop now 😸

Reply:
Thanks, went with your suggestions.  Your idea about the ordering of the args was really smart, I think it is a lot more intuitive now 🥳